website/src/content/service-contract.mdx

```diff
@@ -0,0 +1,102 @@
+
+## Breaking changes
+
+Breaking changes are introduced whenever there is a **major** version release, for example from `v1` to `v2`.
+
+> A new major release does not always mean that breaking changes affect consumer of the packages directly.
+> Because of Semantic Versioning, a breaking change can sometimes mean bumping a version of a dependency package
+
+There are main ways a breaking change can impact a custom application:
+
+1. Affects a component or module _not used_ within the custom application
+2. Affects a component or module _used_ within the custom application
+3. Affects a component or module _used explicitly or implicitly_ interacting with our internal GraphQL APIs
+
+In case of a release with a breaking change of type 1. no action is required. In case of type 2. action is only required if the custom application should make use of the improvement enabled by the breaking change.
+Only in case of 3. commercetools may require a migration to be carried out while continuing to support the outdated version (more below).
+
+All breaking changes are thoroughly documented in the release notes together with migration steps to facilitate the version update. This also helps to identify if changes are needed within the custom application's code.
+
+# Minimal required maintenance
+
+The Merchant Center applications themselves are always kept up-to-date with the released tools and packages used to build custom applications.
+This ensures quality and consistency across our packages while we have to potentially perform migration steps related to a breaking change internally in multiple applications.
+
+We do not expect custom applications receiving the same amount of maintenance as Merchant Center applications.
+However, commercetools recommends that custom applications are **updated at regular periods** and maintained in planned intervals to **not fall back behind too many major versions**.
+This for one enables consumers put new or improved features to use but also ensures that underlying libaries remain up-to-date with potential security patches.
+
+Moreover, fixes or updates will not be backported to old releases. Only the latest release will be continously build upon.
+As a result, we recommend to use tools and services that can be used to get dependency updates and help with the maintenance work. For example Renovate, Dependabot, Greenkeeper, etc.
+
+Rare occasions may require migrating to a major release and commercetools to potentially contact owners of custom applications in the process. This can be caused by changes to internal GraphQL APIs or general required updates for custom applications to ensure reliability and stability.
+As part of the service contract commercetools guarantees that **any major non beta version will be supported for at least 12 months** without requiring maintenance work. Any mandatory breaking release will be documented in the changelogs.
```

Comment:
Can we include a table over AppKit versions and their Support Policy here? Something like;

| Release | Release Date | End of Life Date |
| --- | --- | --- |
| 15.x | 2019-09-02 | 2020-09-02 |
| 14.x | 2019-07-22 | 2020-07-22 |
| 13.x | 2019-04-04 | 2020-04-04 |

Reply:
Hm, seeing how we haven't released a new major version in over a quarter now, maybe be should aim for a bump in major version at least quarterly (whether breaking changes or not) to ensure that customers building an app using the latest version always will have at least 9 months until EOL.

Reply:
> aim for a bump in major version at least quarterly (whether breaking changes or not)

No, I would really advist against that. Version numbers are for computers also and that would break all semantic meaning of that. Even though I understand the intend and purpose behind it. We had a similar argument once of "Can we not bump majors so often. It looks bad.". I think we should follow SemVer here otherwise all automatic tooling kind of breaks of for us.

Reply:
> aim for a bump in major version at least quarterly (whether breaking changes or not)

No, I would really advist against that. Version numbers are for computers also and that would break all semantic meaning of that. Even though I understand the intend and purpose behind it. We had a similar argument once of "Can we not bump majors so often. It looks bad.". I think we should follow SemVer here otherwise all automatic tooling kind of breaks of for us.

Reply:
> Can we include a table over AppKit versions and their Support lifecycle here? Something like;

A table would be nice. Maybe not for beta though cause we wrote that during puplic beta - as with any beta - we don't strictly give the same guarantees.

I would envision something like this. Still, it feels a bit over the top right now (at least to me).

<img width="998" alt="CleanShot 2020-01-08 at 15 59 27@2x" src="https://user-images.githubusercontent.com/1877073/71988273-dd8bcd80-322f-11ea-915a-9239912c6a60.png">

From that one easily also ends up at a release schedule. However, for that we don't have an predictable enough way of working behind the functionality yet from my view.